Gems/Atom/RHI/DX12/Code/Source/RHI/FrameGraphCompiler.cpp

```diff
@@ -564,7 +572,7 @@ namespace AZ
                      // Check if we can service the request at the beginning of the current scope. This is the
                      // normal case when transitioning within the same queue family.
                     if (scopeAfter.IsStateSupportedByQueue(transition.StateBefore) &&
-                        scopeAfter.IsStateSupportedByQueue(transition.StateAfter))
+                        (scopeAfter.GetHardwareQueueClass() == RHI::HardwareQueueClass::Copy ? copyQueueState : scopeAfter.IsStateSupportedByQueue(transition.StateAfter)))
```

Comment:
Why is the check for copy queue is added here? Can you explain what use case is this trying to solve and why cant it be handled by the  IsStateSupportedByQueue function itself?

Reply:
Since this is the only portion of the code that currently could in theory be affected by this state, as this was only set in `FrameGraphCompiler::GetResourceState()` and is only used in this conditional here.
But I understand that this might not be the cleanest solution.
As previously stated, I will reconsider this implementation.